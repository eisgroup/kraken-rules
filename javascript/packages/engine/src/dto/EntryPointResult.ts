/*
 *  Copyright 2018 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { FieldEvaluationResult } from "./FieldEvaluationResult";
import { RuleEvaluationResults } from "./RuleEvaluationResults";
import RuleEvaluationResult = RuleEvaluationResults.RuleEvaluationResult;

import { Reducer } from "declarative-js";
import flat = Reducer.flat;
import { ConditionEvaluationResult } from "./ConditionEvaluationResult";

/**
 * Contains evaluation results from specific entry point
 */
export class EntryPointResult {
    private readonly results: Record<string, FieldEvaluationResult>;
    readonly evaluationTimestamp: Date;
    constructor(results: Record<string, FieldEvaluationResult>) {
        this.results = results;
        this.getApplicableResults.bind(this);
        this.evaluationTimestamp = new Date();
    }
    /**
     * Flatten contents into single list of {@link RuleEvaluationResult}s
     */
    getAllRuleResults(): RuleEvaluationResult[] {
        return Object.values(this.results)
            .map(x => x.ruleResults)
            .reduce(flat, []);
    }
    getFieldResults(): Record<string, FieldEvaluationResult> {
        return this.results;
    }

    getApplicableResults(): RuleEvaluationResult[] {
        return this.getAllRuleResults()
            .filter(r => ConditionEvaluationResult.isApplicable(r.conditionEvaluationResult));
    }

    static empty(): EntryPointResult {
        return new EntryPointResult({});
    }
}
