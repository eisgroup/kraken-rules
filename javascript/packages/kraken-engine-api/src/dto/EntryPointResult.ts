/*
 *  Copyright 2022 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { FieldEvaluationResult } from './FieldEvaluationResult'
import { RuleEvaluationResults } from './RuleEvaluationResults'
import RuleEvaluationResult = RuleEvaluationResults.RuleEvaluationResult

/**
 * Contains evaluation results from specific entry point
 */
export interface EntryPointResult {
    readonly evaluationTimestamp: Date
    readonly ruleTimeZoneId: string

    /**
     * Returns entry point result mapped by field name.
     */
    getFieldResults(): Record<string, FieldEvaluationResult>

    /**
     * Flattens and returns field results into an array of {@link RuleEvaluationResult}
     */
    getAllRuleResults(): RuleEvaluationResult[]

    /**
     * Flattens and returns field results into an array of {@link RuleEvaluationResult}
     * filtered by rule condition evaluation status.
     */
    getApplicableResults(): RuleEvaluationResult[]
}
