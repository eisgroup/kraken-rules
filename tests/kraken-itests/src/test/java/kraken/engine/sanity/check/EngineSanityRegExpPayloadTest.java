/*
 *  Copyright 2019 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package kraken.engine.sanity.check;

import kraken.runtime.engine.EntryPointResult;
import kraken.testproduct.domain.*;
import kraken.utils.MockAutoPolicyBuilder;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;
import static kraken.testing.matchers.KrakenMatchers.*;
import static org.hamcrest.MatcherAssert.assertThat;
import static kraken.test.KrakenItestMatchers.matchesSnapshot;

public final class EngineSanityRegExpPayloadTest extends SanityEngineBaseTest {

    @BeforeClass
    public static void beforeAll() {
        start();
    }

    @AfterClass
    public static void afterAll() {
        validateSnapshots();
    }

    @Test
    public void shouldExecuteRegExpAutoPolicyEntryPointWithEmptyData() {
        Policy policy = new MockAutoPolicyBuilder().addEmptyAutoPolicy().build();
        final EntryPointResult result = engine.evaluate(policy, "RegExpAutoPolicy");

        assertThat(result, hasApplicableResults(5));
        assertThat(result, hasNoValidationFailures());
        assertThat(result, hasNoIgnoredRules());
        assertThat(result, matchesSnapshot());
    }

    @Test
    public void shouldExecuteRegExpAutoPolicyEntryPointWithValidData() {
        Policy policy = new MockAutoPolicyBuilder()
                .addValidAutoPolicyWithMockDateTime()
                .build();

        final EntryPointResult result = engine.evaluate(policy, "RegExpAutoPolicy");

        assertThat(result, hasApplicableResults(11));
        assertThat(result, hasNoValidationFailures());
        assertThat(result, hasNoIgnoredRules());
        assertThat(result, matchesSnapshot());
    }

    @Test
    public void shouldExecuteRegExpAutoPolicyEntryPointWithNotValidData() {
        final BillingInfo billingInfo = new BillingInfo();
        billingInfo.setCreditCardInfo(new CreditCardInfo("MasterCard", "1545464464454545", 132654, null, null));
        final Party party = new Party("1");
        final PersonInfo personInfo = new PersonInfo();
        personInfo.setAddressInfo(new AddressInfo(
                null, null, "123456", "654321"));
        party.setPersonInfo(personInfo);
        Policy policy = new MockAutoPolicyBuilder()
                .addBillingInfo(billingInfo)
                .addPolicyNumber("Not valid")
                .addState("654321")
                .addParty(party)
                .build();

        final EntryPointResult result = engine.evaluate(policy, "RegExpAutoPolicy");

        assertThat(result, hasApplicableResults(7));
        assertThat(result, hasValidationFailures(6));
        assertThat(result, matchesSnapshot());
    }
}
