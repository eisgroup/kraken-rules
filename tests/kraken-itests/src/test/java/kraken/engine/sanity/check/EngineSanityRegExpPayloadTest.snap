kraken.engine.sanity.check.EngineSanityRegExpPayloadTest.shouldExecuteRegExpAutoPolicyEntryPointWithEmptyData=[
  {
    "AddressInfo:-1:city": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "city",
        "fieldPath": "city"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0158",
            "context": "AddressInfo",
            "targetPath": "city",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z ]*$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "AddressInfo:-1:postalCode": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "postalCode",
        "fieldPath": "postalCode"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0157A",
            "context": "AddressInfo",
            "targetPath": "postalCode",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^\\d{5}$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^\\d{5}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^\\d\u0027{\u00275\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "CreditCardInfo:-1:cardNumber": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "CreditCardInfo",
        "fieldName": "cardNumber",
        "fieldPath": "cardNumber"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0062",
            "context": "CreditCardInfo",
            "targetPath": "cardNumber",
            "payloadType": "REGEX"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "NOT_APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "SKIPPED"
        },
        {
          "ruleInfo": {
            "ruleName": "R0063",
            "context": "CreditCardInfo",
            "targetPath": "cardNumber",
            "payloadType": "REGEX"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "NOT_APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "SKIPPED"
        }
      ]
    },
    "CreditCardInfo:-1:cardType": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "CreditCardInfo",
        "fieldName": "cardType",
        "fieldPath": "cardType"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0059",
            "context": "CreditCardInfo",
            "targetPath": "cardType",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z]+$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z]+$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z]+$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "CreditCardInfo:-1:cvv": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "CreditCardInfo",
        "fieldName": "cvv",
        "fieldPath": "cvv"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0066",
            "context": "CreditCardInfo",
            "targetPath": "cvv",
            "payloadType": "REGEX"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "NOT_APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "SKIPPED"
        }
      ]
    },
    "Policy:-1:policyNumber": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "policyNumber",
        "fieldPath": "policyNumber"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0003",
            "context": "Policy",
            "targetPath": "policyNumber",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^Q[0-9]{4}$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^Q[0-9]{4}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^Q[0-9]\u0027{\u00274\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "Policy:-1:state": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "state",
        "fieldPath": "state"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0007",
            "context": "Policy",
            "targetPath": "state",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z]+$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z]+$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z]+$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    }
  }
]


kraken.engine.sanity.check.EngineSanityRegExpPayloadTest.shouldExecuteRegExpAutoPolicyEntryPointWithNotValidData=[
  {
    "AddressInfo:-1:city": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "city",
        "fieldPath": "city"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0158",
            "context": "AddressInfo",
            "targetPath": "city",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z ]*$",
            "success": false,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "AddressInfo:-1:postalCode": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "postalCode",
        "fieldPath": "postalCode"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0157A",
            "context": "AddressInfo",
            "targetPath": "postalCode",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^\\d{5}$",
            "success": false,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^\\d{5}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^\\d\u0027{\u00275\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "CreditCardInfo:-1:cardNumber": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "CreditCardInfo",
        "fieldName": "cardNumber",
        "fieldPath": "cardNumber"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0062",
            "context": "CreditCardInfo",
            "targetPath": "cardNumber",
            "payloadType": "REGEX"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "NOT_APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "SKIPPED"
        },
        {
          "ruleInfo": {
            "ruleName": "R0063",
            "context": "CreditCardInfo",
            "targetPath": "cardNumber",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}$",
            "success": false,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^(?:5[1-5][0-9]\u0027{\u00272\u0027}\u0027|222[1-9]|22[3-9][0-9]|2[3-6][0-9]\u0027{\u00272\u0027}\u0027|27[01][0-9]|2720)[0-9]\u0027{\u002712\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "CreditCardInfo:-1:cardType": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "CreditCardInfo",
        "fieldName": "cardType",
        "fieldPath": "cardType"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0059",
            "context": "CreditCardInfo",
            "targetPath": "cardType",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z]+$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z]+$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z]+$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "CreditCardInfo:-1:cvv": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "CreditCardInfo",
        "fieldName": "cvv",
        "fieldPath": "cvv"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0066",
            "context": "CreditCardInfo",
            "targetPath": "cvv",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[0-9]{3}$",
            "success": false,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[0-9]{3}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[0-9]\u0027{\u00273\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "Policy:-1:policyNumber": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "policyNumber",
        "fieldPath": "policyNumber"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0003",
            "context": "Policy",
            "targetPath": "policyNumber",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^Q[0-9]{4}$",
            "success": false,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^Q[0-9]{4}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^Q[0-9]\u0027{\u00274\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "Policy:-1:state": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "state",
        "fieldPath": "state"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0007",
            "context": "Policy",
            "targetPath": "state",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z]+$",
            "success": false,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z]+$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z]+$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    }
  }
]


kraken.engine.sanity.check.EngineSanityRegExpPayloadTest.shouldExecuteRegExpAutoPolicyEntryPointWithValidData=[
  {
    "AddressInfo:-1:city": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "city",
        "fieldPath": "city"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0158",
            "context": "AddressInfo",
            "targetPath": "city",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z ]*$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "AddressInfo:-1:postalCode": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "postalCode",
        "fieldPath": "postalCode"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0157A",
            "context": "AddressInfo",
            "targetPath": "postalCode",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^\\d{5}$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^\\d{5}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^\\d\u0027{\u00275\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "AddressInfo:2:city": {
      "contextFieldInfo": {
        "contextId": "2",
        "contextName": "AddressInfo",
        "fieldName": "city",
        "fieldPath": "city"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0158",
            "context": "AddressInfo",
            "targetPath": "city",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z ]*$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "AddressInfo:2:postalCode": {
      "contextFieldInfo": {
        "contextId": "2",
        "contextName": "AddressInfo",
        "fieldName": "postalCode",
        "fieldPath": "postalCode"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0157A",
            "context": "AddressInfo",
            "targetPath": "postalCode",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^\\d{5}$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^\\d{5}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^\\d\u0027{\u00275\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "BillingAddress:-1:city": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "BillingAddress",
        "fieldName": "city",
        "fieldPath": "city"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0158",
            "context": "AddressInfo",
            "targetPath": "city",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z ]*$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z ]*$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "BillingAddress:-1:postalCode": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "BillingAddress",
        "fieldName": "postalCode",
        "fieldPath": "postalCode"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0157A",
            "context": "AddressInfo",
            "targetPath": "postalCode",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^\\d{5}$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^\\d{5}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^\\d\u0027{\u00275\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "CreditCardInfo:-1:cardNumber": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "CreditCardInfo",
        "fieldName": "cardNumber",
        "fieldPath": "cardNumber"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0062",
            "context": "CreditCardInfo",
            "targetPath": "cardNumber",
            "payloadType": "REGEX"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "NOT_APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "SKIPPED"
        },
        {
          "ruleInfo": {
            "ruleName": "R0063",
            "context": "CreditCardInfo",
            "targetPath": "cardNumber",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^(?:5[1-5][0-9]\u0027{\u00272\u0027}\u0027|222[1-9]|22[3-9][0-9]|2[3-6][0-9]\u0027{\u00272\u0027}\u0027|27[01][0-9]|2720)[0-9]\u0027{\u002712\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "CreditCardInfo:-1:cardType": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "CreditCardInfo",
        "fieldName": "cardType",
        "fieldPath": "cardType"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0059",
            "context": "CreditCardInfo",
            "targetPath": "cardType",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z]+$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z]+$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z]+$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "CreditCardInfo:-1:cvv": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "CreditCardInfo",
        "fieldName": "cvv",
        "fieldPath": "cvv"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0066",
            "context": "CreditCardInfo",
            "targetPath": "cvv",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[0-9]{3}$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[0-9]{3}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[0-9]\u0027{\u00273\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "Policy:-1:policyNumber": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "policyNumber",
        "fieldPath": "policyNumber"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0003",
            "context": "Policy",
            "targetPath": "policyNumber",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^Q[0-9]{4}$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^Q[0-9]{4}$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^Q[0-9]\u0027{\u00274\u0027}\u0027$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "Policy:-1:state": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "state",
        "fieldPath": "state"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0007",
            "context": "Policy",
            "targetPath": "state",
            "payloadType": "REGEX"
          },
          "payloadResult": {
            "regExp": "^[A-Za-z]+$",
            "success": true,
            "messageCode": "code",
            "message": "String doesn\u0027t match Regular Expression: ^[A-Za-z]+$",
            "templateVariables": [],
            "messageTemplate": "String doesn\u0027\u0027t match Regular Expression: ^[A-Za-z]+$",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    }
  }
]